src/app/store/machine/utils/hooks.ts

```diff
@@ -85,16 +85,21 @@ export const useFetchMachineCount = (
   };
 };
 
+export type UseFetchMachinesOptions = {
+  filters?: FetchFilters | null;
+  grouping?: FetchGroupKey | null;
+  pageSize?: number;
+  currentPage?: number;
+  sortKey?: FetchGroupKey | null;
+  sortDirection?: FetchSortDirection | null;
+  collapsedGroups?: string[];
+};
+
 /**
  * Fetch machines via the API.
  */
 export const useFetchMachines = (
-  filters?: FetchFilters | null,
-  grouping?: FetchGroupKey | null,
-  pageSize?: number,
-  currentPage?: number,
-  sortKey?: FetchGroupKey | null,
-  sortDirection?: FetchSortDirection | null
+  options?: UseFetchMachinesOptions | null
```

Comment:
Passing in an object of options means we don't have to check each param for changes. Is this better do you think? It certainly simplifies the hook and probably means less chance of bugs if we forgot to check one of the params.

Reply:
Absolutely, when the number of arguments becomes larger object makes more sense.